pkg/aws/ec2/api/helper.go

```diff
@@ -594,3 +619,81 @@ func (h *ec2APIHelper) DetachAndDeleteNetworkInterface(attachmentID *string, nwI
 	}
 	return nil
 }
+
+func (h *ec2APIHelper) GetSecurityGroupIdsForSecurityGroupNames(securityGroupNames []string) ([]string, error) {
+	sgIds := make([]string, 0)
+	if len(securityGroupNames) < 1 {
+		return sgIds, nil
+	}
+	sgNamesNotInCache := make([]string, 0)
+	for _, sgName := range securityGroupNames {
+		if sgId, ok := h.securityGroupNameToIdMap[sgName]; !ok {
+			sgNamesNotInCache = append(sgNamesNotInCache, sgName)
+		} else {
+			sgIds = append(sgIds, sgId)
+		}
+	}
+	if len(sgIds) == len(securityGroupNames) {
+		//All SGNames found in cache
+		return sgIds, nil
+	}
+	filters := []*ec2.Filter{
+		{
+			Name:   aws.String("group-name"),
+			Values: aws.StringSlice(sgNamesNotInCache),
+		},
+		{
+			Name:   aws.String("vpc-id"),
+			Values: aws.StringSlice([]string{h.workerNodeVpcId}),
+		},
+	}
+	input := &ec2.DescribeSecurityGroupsInput{Filters: filters}
+	foundSgNames := map[string]bool{}
+	for {
+		output, err := h.ec2Wrapper.DescribeSecurityGroups(input)
+		if err != nil {
+			return nil, err
+		}
+		for _, sg := range output.SecurityGroups {
+			h.securityGroupNameToIdMap[*sg.GroupName] = *sg.GroupId
+			sgIds = append(sgIds, *sg.GroupId)
+			foundSgNames[*sg.GroupName] = true
+		}
+		if output.NextToken == nil {
+			break
+		}
+		input.NextToken = output.NextToken
+	}
+	var missingSgNames []string
+	for _, sgName := range sgNamesNotInCache {
+		if _, ok := foundSgNames[sgName]; !ok {
+			missingSgNames = append(missingSgNames, sgName)
+		}
+	}
+	if len(missingSgNames) > 0 {
+		sort.Strings(missingSgNames)
+		return nil, fmt.Errorf("security groups %s not found", missingSgNames)
+	}
+	return sgIds, nil
+}
+
+// GetSourceAcctAndArn constructs source acct and arn and return them for use
+func GetSourceAcctAndArn(roleARN, region, clusterName string) (string, string, error) {
```

Comment:
Any reason why this is moved from `pkg/utils/helper.go`?

Reply:
There is a circular dependency which required moving a few things out of utils such as this and the httpclient code